### PR TITLE
Fix events on sounds

### PR DESCRIPTION
**Pull request by jabdoa2**

### CODE REVIEW

**Comment by qcapen** [comment]
This fix will work if None is not a setting value that can be overridden in the sound_player (frequently desirable with some settings like max_queue_time).  My original goal with the sound_player was to allow the user to override just about any setting in the sound, including None values.  To do that, I created default values that would make it obvious that they were not provided by the user, but rather were provided by default during the config validation process.  Then the sound_player config validation code could simply delete them from the config.  I really thought this was working properly at one time.  What is strange is it seems those dummy default values (like 'use_sound_setting') would get deleted from the sound_player config during validation, but then were somehow still getting posted during the config player event.  That is the part I could not track down yet.  One easy (but not so elegant) solution is to put the dummy value detection code in the SoundInstance constructor (rather than in the validate_config_item code in the sound_player).  Perhaps I'm overthinking this and not allowing None as an overridden value is just fine.

**Comment by jabdoa2** [comment]
Good point. Maybe we should just change the semantics and disallow "None" as a valid option? max_queue_time 0 makes sense. -1 somehow makes sense but "unlimited" would be better. More me "None" does not have a clear semantic except that is is not set. What would overwriting with "not set" mean? Not sure

**Comment by qcapen** [comment]
I'm going to make a few more changes to this branch to get the behavior I originally had in mind (there will be a corresponding branch in mpf for a few config_spec changes).

**Review by qcapen — APPROVED** [review]
Made some additional changes to restore the original behavior I had intended (ability to override using a None value for settings where it makes sense). There are corresponding changes to the config_spec in mpf (same branch name).

File: mpfmc/assets/sound.py [hunk 1]
```diff
@@ -744,68 +744,65 @@ def __init__(self, sound: Union[SoundAsset, SoundPool],     # noqa
...
-        if 'fade_in' in settings and settings['fade_in'] is not None:
+        if settings.get('fade_in'):
             self._fade_in = settings['fade_in']
 
-        if 'fade_out' in settings and settings['fade_out'] is not None:
+        if settings.get('fade_out'):
             self._fade_out = settings['fade_out']
 
-        if settings.get('about_to_finish_time'):
+        if 'about_to_finish_time' in settings and settings['about_to_finish_time'] != -1:
```
Comment (avanwinkle) [inline mpfmc/assets/sound.py]: Nit: Could be `if settings.get('about_to_finish_time', -1) != -1:`, as with the rest of these settings with default non-`None` values

**Review by avanwinkle — APPROVED** [review]
Cool! I like big blocks of red that clear up old functionality!

File: mpfmc/assets/sound.py (hunk 1 above)
Comment (qcapen) [reply mpfmc/assets/sound.py]: Will do.  Will look prettier anyway :)

**Pull request merged**